common/src/test/java/com/revenuecat/purchases/common/offlineentitlements/OfflineCustomerInfoCalculatorTest.kt

```diff
@@ -154,30 +158,30 @@ class OfflineCustomerInfoCalculatorTest {
 
         // Made a purchase for p1m and then a purchase for not_bw after that one expired
         // You can't have an active purchase of the same product different base plans at the same time
-        val oneHourAgo = 1.hours.ago()
-        val twoHoursAgo = 2.hours.ago()
+        val twoHoursAgo = oneHourAgo.subtract(1.hours)
         val p1mProduct = PurchasedProduct(
             productIdentifier,
             basePlan,
             stubStoreTransactionFromPurchaseHistoryRecord(
                 productIds = listOf(productIdentifier),
                 purchaseTime = twoHoursAgo.time,
             ),
-            false,
+            isActive = false,
```

Comment:
this isActive is for the product, meaning if it comes from the `queryPurchases` call or from `queryPurchasesHistory`. It doesn't mean the entitlement is active or not.

Reply:
Which makes me think... should we be using this for computation of the isActive in the entitlement? 🤔

Reply:
We should test what happens if the device's clock is changed, but I assume the expiration date sent by Google is going to be accurate (device independent). Then rolling back the device clock to before the expiration date will make the entitlement active, if I am not wrong. I think we need to use the fact that the purchase is being returned by queryPurchases or not.

Reply:
Hmm so currently the expiration date is calculated like this:
```
    private fun getExpirationDate(
        isActive: Boolean,
        purchaseAssociatedToProduct: StoreTransaction
    ): Date? {
        return if (purchaseAssociatedToProduct.type == ProductType.SUBS) {
            if (isActive) {
                Date(dateProvider.now.time + TimeUnit.DAYS.toMillis(1))
            } else {
                Date(purchaseAssociatedToProduct.purchaseTime)
            }
        } else {
            null
        }
    }
```

So if I'm not wrong, the requestDate would be greater than the expiration date and we wouldn't grant access. We should still test this to confirm though.

Reply:
Ah wait, ignore my previous comment, you're right if the user moves the device clock, it can indeed change to be active... So yeah, we need to make sure the expiration date we give accounts for moving the device clock. Great find!